core/src/wallet/cosmos/explorers/RpcsParsers.hpp

```diff
@@ -508,6 +508,189 @@ namespace ledger {
+                assert((n.HasMember(kCompletionTime)));
+                assert((n.HasMember(kInitialBalance)));
+                assert((n.HasMember(kBalance)));
+
+                out.creationHeight = BigInt::fromString(n[kCreationHeight].GetString());
+                out.completionTime = DateUtils::fromJSON(n[kCompletionTime].GetString());
+                out.initialBalance = BigInt::fromString(n[kInitialBalance].GetString());
+                out.balance = BigInt::fromString(n[kBalance].GetString());
+            }
+
+            // Parse unbonding result from (/staking/delegators/{address}/unbonding_delegations)
+            // into cosmos::Unbonding
+            // {
+            //   "delegator_address": "cosmos1g84934jpu3v5de5yqukkkhxmcvsw3u2ajxvpdl",
+            //   "validator_address": "cosmosvaloper1clpqr4nrk4khgkxj78fcwwh6dl3uw4epsluffn",
+            //   "entries": [
+            //     {
+            //       "creation_height": "1177125",
+            //       "completion_time": "2020-04-07T10:39:59.296697797Z",
+            //       "initial_balance": "500",
+            //       "balance": "500"
+            //     }
+            //   ]
+            // }
+            template <typename T>
+            void parseUnbonding(const T& n, cosmos::Unbonding &out) {
+                assert((n.HasMember(kDelegatorAddress)));
+                assert((n.HasMember(kValidatorAddress)));
+                assert((n.HasMember("entries")));
+                out.entries = std::list<cosmos::UnbondingEntry>();
+                out.delegatorAddress = n[kDelegatorAddress].GetString();
+                out.validatorAddress = n[kValidatorAddress].GetString();
+                for (const auto& entry : n["entries"].GetArray()) {
+                    cosmos::UnbondingEntry parsedEntry;
+                    parseUnbondingEntry(entry.GetObject(), parsedEntry);
+                    out.entries.emplace_back(std::move(parsedEntry));
+                }
+            }
+
+            // Parse unbonding result from (/staking/delegators/{address}/unbonding_delegations)
+            // into cosmos::UnbondingList
+            // {
+            //   "height": "1347301",
+            //   "result": [
+            //     {
+            //       "delegator_address": "cosmos1g84934jpu3v5de5yqukkkhxmcvsw3u2ajxvpdl",
+            //       "validator_address": "cosmosvaloper1clpqr4nrk4khgkxj78fcwwh6dl3uw4epsluffn",
+            //       "entries": [
+            //         {
+            //           "creation_height": "1177125",
+            //           "completion_time": "2020-04-07T10:39:59.296697797Z",
+            //           "initial_balance": "500",
+            //           "balance": "500"
+            //         }
+            //       ]
+            //     },
+            //     {},
+            //     ...
+            //   ]
+            // }
+            template <typename T>
+            void parseUnbondingList(const T& n, cosmos::UnbondingList &out) {
```

Comment:
Can't this be used in `GaiaCosmosLikeBlockchainExplorer::getUnbondingsByDelegator`?

Reply:
Yup, no idea why I didn't do it